src/runtime_src/core/tools/common/ReportBOStats.cpp

```diff
@@ -0,0 +1,154 @@
+    boost::split(mem_stat, mem_stat_char, boost::is_any_of("\n"));
+    /* Content of mem_stat now will be as below:
+        [UNUSED] bank0@0x004000000000 (16384MB): 0KB 0BOs
+        [IN-USE] bank1@0x005000000000 (16384MB): 0KB 0BOs
+        [UNUSED] bank2@0x006000000000 (16384MB): 0KB 0BOs
+        [UNUSED] bank3@0x007000000000 (16384MB): 0KB 0BOs
+        [UNUSED] PLRAM[0]@0x003000000000 (0MB): 0KB 0BOs
+        [UNUSED] PLRAM[1]@0x003000400000 (0MB): 0KB 0BOs
+        [UNUSED] PLRAM[2]@0x003000600000 (0MB): 0KB 0BOs
+        [IN-USE] HOST[0]@0x002000000000 (16384MB): 0KB 0BOs
+        [== BO Stats Below ==] NA@0x000000000000 (0MB): 0KB 0BOs
+        [Regular] 0KB 0BOs
+        [UserPointer] 0KB 0BOs
+        [P2P] 0KB 0BOs
+        [DeviceOnly] 0KB 0BOs
+        [Imported] 0KB 0BOs
+        [ExecBuf] 0KB 0BOs
+        [CMA] 0KB 0BOs
+    */
+    bool found = false;
+
+    //Skip DDR usage info by looking for line with string "BO Stats Below"
+    //Following that are BO stats. BO type, size in KB and num of BOs fields separated by blank space
+    for (const auto& line: mem_stat) {
+      if (line.empty())
+        continue;
+
+      //Skip DDR usage stats at begining by looking for beloe string
+      if (!found && line.find("BO Stats Below") != std::string::npos) {
+        found = true;
+	continue;
+      } else if (!found)
+	continue;
+
+      //Capture BOStat fields separated by blank space
+      std::vector<std::string> bo_info;
+      boost::split(bo_info, line, boost::is_any_of("\t "));
+      //Expected fileds are BO type, size in KB and Num of BOs
+      if (bo_info.size() != 3)
+        throw std::runtime_error((boost::format("ERROR: Unexpected format in BO Stats. Line: %s") % line).str());
+
+      boost::property_tree::ptree bo_pt;
+      if (bo_info[0].length() < 5 || bo_info[1].length() < 3 || bo_info[2].length() < 4) {
+        bo_pt.put("bufer_type", "Invalid");
+        bo_pt.put("buffer_count", "n/a");
+        bo_pt.put("memory_used", "n/a");
+        bo_pt.put("memory_used_unit", "n/a");
+        bo_pt.put("memory_used_bytes", "n/a");
+      } else {
+	std::string mem_used(bo_info[1]);
+	mem_used.pop_back();
+	auto mem_used_bytes = XBUtilities::string_to_bytes(mem_used);
+	mem_used.pop_back();
+        bo_pt.put("buffer_type", bo_info[0].substr(1, bo_info[0].length()-2));
+        bo_pt.put("buffer_count", bo_info[2].substr(0, bo_info[2].length()-3));
+        bo_pt.put("memory_used", mem_used);
+        bo_pt.put("memory_used_unit", bo_info[1].substr(bo_info[1].length()-2, 2));
+        bo_pt.put("memory_used_bytes", mem_used_bytes);
+      }
+      pt2_list.push_back(std::make_pair("", bo_pt));
+    }
+  } catch(const xrt_core::error& e) {
+    std::cerr << "ERROR: " << e.what() << std::endl;
+    std::cerr << "ERROR: bo_stats cmd - invalid format" << std::endl;
+    return; //Give error and let script continue

```

Comment:
xbmgmt and xbutil returns a non-zero value whenever an error occurs.   This code masks this functionality.

Please throw an exception.

Reply:
done